src/Entity/InformationProduct.php

```diff
@@ -414,6 +428,46 @@ public function removeInformationProductType(InformationProductTypeDescriptor $i
         $this->informationProductTypeDescriptors->removeElement($informationProductTypeDescriptor);
     }
 
+    /**
+     * Get all Digitial Resource Type Descriptors.
+     *
+     * @return Collection|DigitalResourceTypeDescriptor[]
+     */
+    public function getDigitalResourceTypeDescriptors(): Collection
+    {
+        return $this->digitalResourceTypeDescriptors;
+    }
+
+    /**
+     * Add Digital Resource Type Descriptor to Information Product.
+     *
+     * @param DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor
+     *
+     * @return self
+     */
+    public function addDigitalResourceTypeDescriptor(DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor): self
+    {
+        if (!$this->digitalResourceTypeDescriptors->contains($digitalResourceTypeDescriptor)) {
+            $this->digitalResourceTypeDescriptors[] = $digitalResourceTypeDescriptor;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Remove Digital Resource Type Descriptor to Information Product.
+     *
+     * @param DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor
+     *
+     * @return self
+     */
+    public function removeDigitalResourceTypeDescriptor(DigitalResourceTypeDescriptor $digitalResourceTypeDescriptor): self
+    {
+        $this->digitalResourceTypeDescriptors->removeElement($digitalResourceTypeDescriptor);
+
+        return $this;
```

Comment:
Do we need the return on this?

Reply:
I like the chaining, so I think we should keep it.